akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -757,11 +786,38 @@ abstract class ShardCoordinator(
               self.tell(GetShardHome(shard), ignoreRef)
             }
           } else {
-            // rebalance not completed, graceful shutdown will be retried
-            gracefulShutdownInProgress -= state.shards(shard)
             clearRebalanceInProgress(shard)
           }
         } else {
+          log.warning("Rebalance shard [{}] didn't complete within [{}].", shard, handOffTimeout.pretty)
+          clearRebalanceInProgress(shard)
+        }
```

Comment:
The handling of `RebalanceDone` doesn't need to take the graceful shutdown into account. So, the if and else logic is simplified.